openhgnn/models/MAGNN.py

```diff
@@ -119,24 +123,58 @@ def __init__(self, in_feats, h_feats, inter_attn_feats, num_heads, num_classes,
         self.feat_drop = nn.Dropout(p=dropout_rate)
 
         # extract ntypes that have corresponding metapath
-        # If there're only metapaths like ['MAM', 'MDM'], 'A' and 'D' have no metapath.
-        self.meta_ntypes = set([metapath[0] for metapath in metapath_list])
+        # If there're only metapaths like ['MAM', 'MDM'], 'A' and 'D' have no metapath, so that 'A' and 'D' shouldn't
+        # be considered as nodes that need to aggregate information from metapath.
+        self.dst_ntypes = set([metapath[0] for metapath in metapath_list])
 
         # hidden layers
         self.layers = nn.ModuleList()
         for i in range(num_layers - 1):
             self.layers.append(
                 MAGNN_layer(in_feats=h_feats, inter_attn_feats=inter_attn_feats, out_feats=h_feats, num_heads=num_heads,
                             metapath_list=metapath_list, ntypes=self.ntypes, edge_type_list=edge_type_list,
-                            meta_ntypes=self.meta_ntypes, encoder_type=encoder_type, last_layer=False))
+                            dst_ntypes=self.dst_ntypes, encoder_type=encoder_type, last_layer=False))
 
         # output layer
         self.layers.append(
             MAGNN_layer(in_feats=h_feats, inter_attn_feats=inter_attn_feats, out_feats=num_classes, num_heads=num_heads,
                         metapath_list=metapath_list, ntypes=self.ntypes, edge_type_list=edge_type_list,
-                        meta_ntypes=self.meta_ntypes, encoder_type=encoder_type, last_layer=True))
+                        dst_ntypes=self.dst_ntypes, encoder_type=encoder_type, last_layer=True))
 
-        self.metapath_idx_dict = mp_instances
+        self.metapath_idx_dict = metapath_idx_dict
+
+    def mini_reset_params(self, new_metapth_idx_dict):
```

Comment:
Where is this method used?

Reply:
Because we need to reset some parameters (like metapath instances etc) of model based on the subgraph in each iteration. `mini_reset_params()` is used to do this. I'll commit a file containing the training codes utilizing MAGNN sampler. U can see how this method is used in the file.

Reply:
Then could you upload the file together in this PR?  Otherwise I would probably forget.